### PR TITLE
ceph-daemon: a couple fixes

### PR DESCRIPTION
**Pull request by liewegas**
- fix deploy (get_config_and_both_keyrings return value was broken)
- rm-clsuter now removes units even if they are in failed state

### CODE REVIEW

File: src/ceph-daemon [hunk 1]
```diff
@@ -316,7 +316,7 @@ def get_config_and_both_keyrings():
             with open(args.config_and_keyrings, 'r') as f:
                 j = f.read()
         d = json.loads(j)
-        (d.get('config'), d.get('keyring'), d.get('crash_keyring'))
+        return (d.get('config'), d.get('keyring'), d.get('crash_keyring'))
```
Comment (sebastian-philipp) [inline src/ceph-daemon]: And this is the very reason why we need to make this a proper Python package! It's not just about splitting a big Python file. Its about Python tools in the Python ecosystem are build around proper python packages: stup.py, pytest, pytest-mypy, tox, etc. Preventing these type errors is easy if we incorporate proper tools into this project. And especially in Python we need these tools; much more that in the C++ land.

Python need to be a first class citizen in the Ceph project, if we don't want to ship these kind of bugs, like we do today. That's why I'm e.g. building python-common as a proper Python package.

Comment (liewegas) [reply src/ceph-daemon]: Is it not possible to write tests when it's a single file?  This was trivially caught the first time I ran the script; it seems like the problem is a lack of any tests, not the kind of tests.

Comment (liewegas) [reply src/ceph-daemon]: It also seems like the kind of tests we primarily need are the end-to-end functional variety that actually run podman/docker and spin up ceph daemons, not unit tests.

Comment (sebastian-philipp) [reply src/ceph-daemon]: > Is it not possible to write tests when it's a single file?

Sure it is. But adding them should be easy, not hard. Too often I got the response that adding tests is too hard.

> This was trivially caught the first time I ran the script; it seems like the problem is a lack of any tests, not the kind of tests.

Let's leverage tools for that mundane work. I don't want to write functional tests for all code paths just to find those trivial bugs.

> It also seems like the kind of tests we primarily need are the end-to-end functional variety that actually run podman/docker and spin up ceph daemons, not unit tests.

Disagreeing here.

**Pull request merged**